xalia/UiDom/UiDomRoot.cs

```diff
@@ -25,5 +27,16 @@ internal void RaiseElementDiedEvent(UiDomElement element)
         {
             Application.ElementDied(element);
         }
+
+        public void AddGlobalProvider(IUiDomProvider provider, int index)
+        {
+            GlobalProviders.Insert(index, provider);
+            AddedGlobalProvider(provider);
+        }
+
+        public void AddGlobalProvider(IUiDomProvider provider)
+        {
+            AddProvider(provider, Providers.Count);
+        }

```

Comment:
Similarly, I assume this should call `AddGlobalProvider(provider, GlobalProviders.Count);`